components/browser/storage-sync-places/src/main/java/mozilla/components/browser/storage/places/PlacesHistoryStorage.kt

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.storage.places
+
+import android.content.Context
+import android.support.annotation.VisibleForTesting
+import kotlinx.coroutines.experimental.async
+import kotlinx.coroutines.experimental.CoroutineScope
+import kotlinx.coroutines.experimental.Deferred
+import kotlinx.coroutines.experimental.Dispatchers
+import kotlinx.coroutines.experimental.launch
+import mozilla.components.concept.storage.HistoryStorage
+import mozilla.components.concept.storage.PageObservation
+import mozilla.components.concept.storage.VisitType
+import org.mozilla.places.PlacesAPI
+import org.mozilla.places.VisitObservation
+
+class PlacesHistoryStorage(context: Context)
+    : TestablePlacesHistoryStorage(Conn.getInstance(context.getDatabasePath(DB_NAME).canonicalPath))
```

Comment:
What exactly is the reason for `TestablePlacesHistoryStorage`? I assume that you do not want to make `PlacesHistoryStorage` open. If this is just the case for mocking then creating this file in the component will allow you to just mock the non-open class: https://github.com/mozilla-mobile/android-components/blob/master/components/browser/engine-gecko/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker